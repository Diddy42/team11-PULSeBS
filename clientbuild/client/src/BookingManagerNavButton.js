import React from 'react';
import {Link} from 'react-router-dom';

class BookingManagerNavButton extends React.Component{
    constructor(props){
        super(props);

        this.state = { onStatisticsPage: true };
    }

    render(){
        return <>
            {
                this.state.onStatisticsPage &&
                <Link className='mr-4 text-light' to='/booking-manager/contact-tracing' onClick={this.handleClick}>
                    Generate contact tracing report
                </Link>
            }

            {
                !this.state.onStatisticsPage &&
                <Link className='mr-4 text-light' to='/booking-manager' onClick={this.handleClick}>
                    Statistics page
                </Link>
            }
        </>
    }

    handleClick = () => {
        this.setState({ onStatisticsPage: !this.state.onStatisticsPage });
    }
}

export default BookingManagerNavButton;