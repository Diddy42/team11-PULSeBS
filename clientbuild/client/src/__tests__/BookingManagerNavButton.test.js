import TestRenderer from 'react-test-renderer';
import React from 'react';
import BookingManagerNavButton from '../BookingManagerNavButton';
import { shallow } from 'enzyme';
import {BrowserRouter as Router} from 'react-router-dom';

import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

test('render button', () => {
    const tree = TestRenderer.create(<Router><BookingManagerNavButton/></Router>).toJSON();
    expect(tree).toMatchSnapshot();
  });


test('handleClick', async () => {
  const component = shallow(<BookingManagerNavButton/>);
  const instance = component.instance();
  expect(instance.state.onStatisticsPage).toBe(true);
  await instance.handleClick();
  expect(instance.state.onStatisticsPage).toBe(false);
});