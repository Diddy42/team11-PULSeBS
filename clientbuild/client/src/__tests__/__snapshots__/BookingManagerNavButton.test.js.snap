// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render button 1`] = `
<a
  className="mr-4 text-light"
  href="/booking-manager/contact-tracing"
  onClick={[Function]}
>
  Generate contact tracing report
</a>
`;
